sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala

```diff
@@ -424,4 +424,17 @@ class QueryExecutionErrorsSuite
       matchMsg = true
     )
   }
+
+  test("UNSUPPORTED_SAVE_MODE: unsupported null saveMode") {
+    withTempPath { path =>
+      val e = intercept[SparkIllegalStateException] {
+        val saveMode: SaveMode = null
```

Comment:
Besides of passing a null, is it possible to trigger the error from user space?

Reply:
SaveMode only have four value,
https://github.com/apache/spark/blob/master/sql/core/src/main/java/org/apache/spark/sql/SaveMode.java#L27~L57
Append, Overwrite, ErrorIfExists, Ignore

According to the following code, only can trigger QueryExecutionErrors.unsupportedSaveModeError when SaveMode = null.
https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala#L114~L138